src/mono/wasm/runtime/diagnostics/server_pthread/ipc-protocol/base-parser.ts

```diff
@@ -113,7 +113,23 @@ const Parser = {
             result[i] = (buf[j + 2 * i + 1] << 8) | buf[j + 2 * i];
         }
         advancePos(pos, length * 2);
-        return String.fromCharCode.apply(null, result);
+
+        /* Trim trailing nul character(s) that are added by the protocol */
+        let trailingNulStart = -1;
+        if (result.length > 0) {
```

Comment:
i don't think this `if` is necessary since the loop termination condition will bail out